File: app/src/main/java/ru/kireev/mir/laregistrationmain/data/VolunteersDatabase.java
package ru.kireev.mir.laregistrationmain.data;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {Volunteer.class}, version = 2, exportSchema = false)
public abstract class VolunteersDatabase extends RoomDatabase {
    private static VolunteersDatabase database;
    private static final String DB_NAME = "volunteers.db";
    private static final Object LOCK = new Object();

    public static VolunteersDatabase getInstance(Context context) {
        synchronized (LOCK) {
            if (database == null) {
                database = Room.databaseBuilder(context, VolunteersDatabase.class, DB_NAME)
                        .fallbackToDestructiveMigration()
                        .build();
            }
        }
        return database;
    }

    public abstract VolunteersDao volunteersDao();
}

File: app/src/main/java/ru/kireev/mir/registrarlizaalert/data/VolunteersDatabase.kt
package ru.kireev.mir.registrarlizaalert.data

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [Volunteer::class], version = 1, exportSchema = false)
abstract class VolunteersDatabase : RoomDatabase() {
    companion object {
        private var db: VolunteersDatabase? = null
        private const val DB_NAME = "main.db"
        private val LOCK = Any()

        fun getInstance(context: Context): VolunteersDatabase {
            synchronized(LOCK) {
                db?.let { return it }
                val instance = Room.databaseBuilder(
                        context,
                        VolunteersDatabase::class.java,
                        DB_NAME)
                        .build()
                db = instance
                return instance
            }
        }
    }

    abstract fun volunteersDao(): VolunteersDao

}